src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -888,6 +905,79 @@ internal void ExecuteSubmission(BuildSubmission submission, bool allowMainThread
             }
         }
 
+        /// <summary>
+        /// This method adds the graph build request in the specified submission to the set of requests being handled by the scheduler.
+        /// </summary>
+        internal void ExecuteGraphSubmission(BuildSubmission submission)
+        {
+            lock (_syncLock)
+            {
+                VerifyStateInternal(BuildManagerState.Building);
+
+                try
+                {
+                    // Create the build request
+                    submission.BuildRequest = new BuildRequest(
+                        submission.SubmissionId,
+                        BackEnd.BuildRequest.InvalidNodeRequestId,
+                        GetNewConfigurationId(), // TODO: Should we create a real BuildRequestConfiguration for each graph request?
```

Comment:
`ExecuteSubmission` seems to create a unique configuration id for the project. Should I create a unique one for the graph? Or does it even matter in this case (I don't believe we will ever look this up by configuration id)?

Reply:
After a short poking around it seems that if a configuration ID starts flowing through the engine, it is kind of assumed it will actually exist in the ConfigCache. Assigning an id for an inexistent BuildRequestConfiguration might lead to funky behaviour. But I'm not super sure about that. 

---
In reply to: [227636190](https://github.com/Microsoft/msbuild/pull/3879#discussion_r227636190) [](ancestors = 227636190)

Reply:
After reading further into the PR, it seems that you actually never use this BuildSubmission, so I guess you could just set the invalid id on it.

---
In reply to: [227900171](https://github.com/Microsoft/msbuild/pull/3879#discussion_r227900171) [](ancestors = 227900171,227636190)